frontend/src/app/services/resolver-servis/resolver.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, Router, RouterStateSnapshot } from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { RouteName } from 'src/app/enums/route-name-enun';
+import { Product } from 'src/app/interfaces/product';
+import { ProductService } from '../product/product.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ResolverService implements Resolve<Product> {
+  constructor(
+    private productService: ProductService,
+    private router: Router,
+  ) { }
+
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<Product> {    
+    let product    
+    this.productService.getProductsById(+route.params['id'])
+      .subscribe((data: Product) => product = data);
+
+    if(!product) {
```

Comment:
It is not working code. You will get always false in this condition if you will use real server.
This code is async